src/main/resources/META-INF/rewrite/java-version-17.yml

```diff
@@ -59,6 +59,7 @@ recipeList:
   - org.openrewrite.java.migrate.SunNetSslPackageUnavailable
   - org.openrewrite.java.migrate.RemovedRMIConnectorServerCredentialTypesConstant
   - org.openrewrite.java.migrate.RemovedFileIOFinalizeMethods
+  - org.openrewrite.java.migrate.RemovedToolProviderConstructor
```

Comment:
Did you explore using `org.openrewrite.java.ChangeMethodTargetToStatic` here instead of a custom visitor?
https://docs.openrewrite.org/recipes/java/changemethodtargettostatic

You can see here that it also supports removing constructor calls before a method invocation.
https://github.com/openrewrite/rewrite/blob/ed6f97a68d58703d689cdad54f05b5c0d4ee54ca/rewrite-java-test/src/test/java/org/openrewrite/java/ChangeMethodTargetToStaticTest.java#L50-L68

Reply:
Hi @timtebeek I couldn't use the above recipe because the following condition wasn't met Hence I had to create  this recipe. 
```
 if ((!isStatic || !isSameReceiverType) && methodMatcher.matches(method, matchUnknownTypes)) {
```

Reply:
I've gone ahead and changed the `ChangeMethodTargetToStatic` in openrewrite/rewrite; seemed like a common enough use case to fix once and reduce this down to just a declarative recipe for simplicity.
- https://github.com/openrewrite/rewrite/pull/4201